src/cmd/cloud.js

```diff
@@ -793,45 +806,57 @@ module.exports = class CloudCommand extends CLICommandBase {
 			'**/*.ino',
 			'**/*.cpp',
 			'**/*.c',
+			'**/build.mk',
 			'project.properties'
 		];
 
-		if (fs.existsSync(includesFile)){
-			//grab and process all the files in the include file.
+		const result = utilities.globList(dirname, includes, { followSymlinks });
+		result.forEach((file) => files.add(file));
+	}
 
-			includes = utilities.trimBlankLinesAndComments(
-				utilities.readAndTrimLines(includesFile)
-			);
-			hasIncludeFile = true;
+	_getCustomIncludes(files, dirname, { followSymlinks }) {
+		const includeFiles = utilities.globList(dirname, ['**/particle.include'], { followSymlinks });
+		const result = [];
 
+		for (const includeFile of includeFiles) {
+			const includeDir = path.dirname(includeFile);
+			const globsToInclude = utilities.trimBlankLinesAndComments(utilities.readAndTrimLines(includeFile));
+			if (!globsToInclude || !globsToInclude.length) {
+				continue;
+			}
+			const globList = globsToInclude.map(g => g);
+			const includePaths = utilities.globList(includeDir, globList, { followSymlinks });
+			result.push(...includePaths);
```

Comment:
You could just do `includePaths.forEach((file) => files.add(file));` instead of putting them in another temporary array.